packages/core/test/install/patch.ts

```diff
@@ -96,6 +96,28 @@ test('patch package', async () => {
   expect(fs.readFileSync('node_modules/is-positive/index.js', 'utf8')).not.toContain('// patched')
 })
 
+test('patch package reports warning if not all patches are applied and allowNonAppliedPatches is set', async () => {
+  prepareEmpty()
+  const patchPath = path.join(f.find('patch-pkg'), 'is-positive@1.0.0.patch')
+
+  const patchedDependencies = {
+    'is-positive@1.0.0': path.relative(process.cwd(), patchPath),
+    'is-negative@1.0.0': path.relative(process.cwd(), patchPath),
+  }
+  const opts = await testDefaults({
+    fastUnpack: false,
+    sideEffectsCacheRead: true,
+    sideEffectsCacheWrite: true,
+    patchedDependencies,
+    allowNonAppliedPatches: true,
+  }, {}, {}, { packageImportMethod: 'hardlink' })
+  await install({
+    dependencies: {
+      'is-positive': '1.0.0',
+    },
+  }, opts)
+})
+
```

Comment:
you're not testing that the warning is reported. Check how tests spy on the "reporter"

Reply:
Thank you for the tip. For my note (or others looking to do something similar) I used [storePrune.ts ](https://github.com/pnpm/pnpm/blob/17e69e18b7fd4d260ef0e7b3698cd08d1fcdfb95/packages/plugin-commands-store/test/storePrune.ts#L228)as an example

Reply:
The test failed.

Reply:
fixed

Reply:
:heart: Thank you for fixing it for me!